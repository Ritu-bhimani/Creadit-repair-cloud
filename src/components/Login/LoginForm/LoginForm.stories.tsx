import { ComponentMeta, ComponentStory } from '@storybook/react';
import { LoginForm } from './LoginForm';

export default {
  title: 'Login Form',
  component: LoginForm
} as ComponentMeta<typeof LoginForm>;

const Template: ComponentStory<typeof LoginForm> = (args) => (
  <LoginForm {...args} />
);

export const Primary = Template.bind({});

Primary.args = {};
