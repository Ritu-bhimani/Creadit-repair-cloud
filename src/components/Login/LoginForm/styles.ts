import { css } from '@emotion/react';

export const LoginFormStyles = {
  root: css`
    text-align: center;
  `,
  loginForm: css`
    max-width: 400px;
    width: 90%;
    margin: 2vh auto 0;
    border: 1px solid #dedede;
    padding: 10px 30px;
    box-sizing: border-box;
  `,
  field: css`
    background: #fff;
    border: 1px solid #ccc;
    color: #333;
    font-size: 18px;
    font-family: Arial, Helvetica, sans-serif;
    margin: 3px 0;
    padding: 4px 10px;
    width: 100%;
    box-sizing: border-box;
    height: 43px;
    margin-bottom: 15px;
    border-radius: 0;
    :focus-visible{
      outline:none;
    }
    :focus{
      border-color:#00aeef;
    }
  `,
  loginBtn: css`
    :hover {
      background: #0083d6;
      border-color: #093973;
      cursor: pointer;
    }
    border: 1px solid #00aeef;
    color: #fff;
    font-weight: 400;
    width: 100%;
    font-size: 20px;
    line-height:40px;
    text-align: center;
    background: #00aeef;
    margin: 25px 0 0;
  `,
  greeting: css`
    color: #6b7a8d;
    display: block;
    font-size: 2em !important;
    font-family: Arial, Helvetica, sans-serif !important;
    margin-block-start: 0.67em !important;
    margin-block-end: 0.67em !important;
    margin-inline-start: 0px !important;
    margin-inline-end: 0px !important;
    font-weight: bold !important;
  `,
  forgotPassword: css`
    clear: both;
    display: block;
    padding: 10px 20px;
    margin-top: 10px;
    color: #478cd3;
    text-decoration: none;
    font-weight: 700;
    font-size:14px;
    :hover{
      text-decoration:underline;
    }
  `,
  errorMessage: css`
    color: red !important;
  `
};
