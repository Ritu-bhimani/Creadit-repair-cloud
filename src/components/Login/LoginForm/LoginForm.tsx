import { Container, Typography } from '@mui/material';
import { ErrorMessage, Field, Form, Formik } from 'formik';
import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import * as yup from 'yup';
import { LoginFormStyles } from './styles';

export type LoginFormProps = {
  onSubmit: (values: any) => void;
  showModal?: any;
};

// Pattern for password validation
const PASSWORD_PATTERN =
  /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/;

export const LoginForm: FC<LoginFormProps> = (props: LoginFormProps) => {
  // Translation hook
  const { t } = useTranslation();
  const { onSubmit, showModal } = props;

  const PASSWORD_PATTERN_WARNING = t('passwordPatternInvalid');
  const INVALID_EMAIL_TEXT = t('invalidEmail');
  const EMAIL_REQUIRED_TEXT = t('emailIdRequired');
  const PASSWORD_REQUIRED_TEXT = t('passwordRequired');

  // Email & Password Validation Schema for Formik
  const validationSchema = yup.object({
    email: yup.string().email(INVALID_EMAIL_TEXT).required(EMAIL_REQUIRED_TEXT),
    password: yup
      .string()
      .required(PASSWORD_REQUIRED_TEXT)
  });
  return (
    <Container css={LoginFormStyles.root}>
      <Formik
        initialValues={{ email: '', password: '' }}
        validationSchema={validationSchema}
        onSubmit={(values, { setSubmitting }) => {
          onSubmit(values);
          setSubmitting(false);
        }}
      >
        {({ isSubmitting }) => (
          <Form css={LoginFormStyles.loginForm}>
            <Typography css={LoginFormStyles.greeting} variant="h2">
              {t('LoginHello')}
            </Typography>
            <Field type="email" name="email">
              {({
                field,
                form: { isSubmitting }
              }: {
                field: any;
                form: { isSubmitting: boolean };
              }) => (
                <input
                  css={LoginFormStyles.field}
                  {...field}
                  disabled={isSubmitting}
                  type="text"
                  placeholder={t('LoginEmailPlaceHolder')}
                />
              )}
            </Field>
            <ErrorMessage
              name="email"
              component="div"
              css={LoginFormStyles.errorMessage}
            />
            <Field type="password" name="password">
              {({
                field,
                form: { isSubmitting }
              }: {
                field: any;
                form: { isSubmitting: boolean };
              }) => (
                <input
                  css={LoginFormStyles.field}
                  {...field}
                  disabled={isSubmitting}
                  type="password"
                  placeholder={t('LoginPasswordPlaceHolder')}
                />
              )}
            </Field>
            <ErrorMessage
              name="password"
              component="div"
              css={LoginFormStyles.errorMessage}
            />
            <button
              css={LoginFormStyles.loginBtn}
              type="submit"
              disabled={isSubmitting}
            >
              {t('Login')}
            </button>
            <a
              css={LoginFormStyles.forgotPassword}
              id="forgotpassword"
              href="#"
              onClick={() => showModal(true)}
            >
              {t('LoginPassword')}
            </a>
          </Form>
        )}
      </Formik>
    </Container>
  );
};
