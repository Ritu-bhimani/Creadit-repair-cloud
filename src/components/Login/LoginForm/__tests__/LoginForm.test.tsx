import { render } from '@testing-library/react';
import { LoginForm } from '../LoginForm';

it('should render and match the snapshot', async () => {
  const component = render(<LoginForm onSubmit={(): void => {}} />);
  expect(component).toMatchSnapshot();
});
