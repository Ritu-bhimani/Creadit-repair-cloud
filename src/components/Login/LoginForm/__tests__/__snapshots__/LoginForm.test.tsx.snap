// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <form
          action="#"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            LoginHello
          </h2>
          <input
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            name="email"
            placeholder="LoginEmailPlaceHolder"
            type="text"
            value=""
          />
          <input
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            name="password"
            placeholder="LoginPasswordPlaceHolder"
            type="password"
            value=""
          />
          <button
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            type="submit"
          >
            Login
          </button>
          <a
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            href="#"
            id="forgotpassword"
          >
            LoginPassword
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <form
        action="#"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          LoginHello
        </h2>
        <input
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          name="email"
          placeholder="LoginEmailPlaceHolder"
          type="text"
          value=""
        />
        <input
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          name="password"
          placeholder="LoginPasswordPlaceHolder"
          type="password"
          value=""
        />
        <button
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          type="submit"
        >
          Login
        </button>
        <a
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          href="#"
          id="forgotpassword"
        >
          LoginPassword
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
